### ISSUE TITLE
filtering a react-infinite-tree still renders blank lines for filtered nodes

### ISSUE THREAD

**Issue by firetiger77** [cheton/infinite-tree#15]
Great library! Forgive me if I'm missing something obvious, but even when using the rowRenderer below, the tree still shows blank lines for the nodes filtered out. This results in a tree with only the filtered nodes (correct), but spaced by large amounts of blank rows. It seems that even returning an empty string produces a row in the tree.

rowRenderer={({ node, tree}) => {
                if (node.state.filtered === false) {
                    return('');
                }
...

**Comment by cheton** [comment]
Empty rows should be filtered out when returning _null_, _undefined_, _false_, or an empty string in your renderer function.

https://github.com/cheton/infinite-tree/blob/master/src/infinite-tree.js#L1564-L1576
```js
update() {
    // Emit a "contentWillUpdate" event
    this.emit('contentWillUpdate');

    if (this.clusterize) {
        // Update list
        const rows = this.rows.filter(row => !!row);
        this.clusterize.update(rows);
    }

    // Emit a "contentWillUpdate" event
    this.emit('contentDidUpdate');
}
```

You can try below example to check if it's working by your side, or provide your example on CodePen.
https://github.com/cheton/infinite-tree/blob/master/examples/classic/index.js#L263-L315
https://github.com/cheton/infinite-tree/blob/master/examples/classic/renderer.js

**Comment by cheton** [comment]
This issue should be belong to react-infinite-tree. I recently made a breaking change to react-infinite-tree, now it requires a different way for filtering nodes. I will make necessary changes to the code to make filter works.

**Comment by cheton** [comment]
It's fixed in [react-infinite-tree@1.0.0-alpha.2](https://github.com/cheton/react-infinite-tree/releases/tag/v1.0.0-alpha.2)

Now you don't need to return empty string in **rowRenderer** or return zero height in **rowHeight**, it's now internally processed by react-infinite-tree.


https://github.com/cheton/react-infinite-tree/tree/master#rendering-tree-nodes
```js
<InfiniteTree
    width="100%"
    height={400}
    rowHeight={30}
    data={props.data}
>
{({ node, tree }) => {
    // Same effect as using rowRenderer
}}
</InfiniteTree>
```

**Comment by firetiger77** [comment]
The fix works well, although I've noticed I still need to pass in a rowRenderer; the nomenclature above doesn't seem to work for me.
